File: gfapy/line/edge/containment/pos.py
import gfapy

class Pos:

  @property
  def rpos(self):
    """The rightmost coordinate of the contained sequence in the container.

    Returns:
      int : 0-based coordinate

    Raises:
      gfapy.ValueError : If the overlap is a placeholder, thus the computation
        cannot be performed.
    """
    if isinstance(self.overlap, gfapy.Placeholder):
      raise gfapy.ValueError("The overlap is a placeholder, therefore"+
          "rpos cannot be computed")
    return self.pos + self.overlap.length_on_reference()

File: lib/rgfa/line/edge/containment/pos.rb
module RGFA::Line::Edge::Containment::Pos

  # Computes the rightmost coordinate of the contained sequence in the container
  # @return [Integer] 0-based right coordinate of contained in container
  # @raise [RGFA::ValueError] if the overlap is not a CIGAR string
  def rpos
    raise RGFA::ValueError if overlap.kind_of?(RGFA::Placeholder)
    pos + overlap.length_on_reference
  end

end
